network-config-analyzer/NetworkConfigQuery.py

```diff
@@ -1144,12 +1194,9 @@ def exec(self):
         if query_answer.output_result:
             return query_answer  # non-identical configurations are not comparable
 
-        if self.config1.type == NetworkConfig.ConfigType.Ingress \
-                or self.config2.type == NetworkConfig.ConfigType.Ingress:
-            ingress_name = self.config1.name if self.config1.type == NetworkConfig.ConfigType.Ingress \
-                else self.config2.name
-            return QueryAnswer(bool_result=False,
-                               output_result=f'Ignoring PermitsQuery for {ingress_name} with Ingress only',
+        if self.config1.layers.does_contain_single_layer(NetworkLayerName.Ingress) or \
+                self.config2.layers.does_contain_single_layer(NetworkLayerName.Ingress):
+            return QueryAnswer(bool_result=False, output_result='Ignoring PermitsQuery for config with Ingress only',
```

Comment:
As we discussed, it's only the **definition** of permitted traffic for which we don't want to use Ingress

Reply:
done